src/test/java/com/exactpro/th2/FixHandlerTest.java

```diff
@@ -225,6 +225,41 @@ void onConnectionTest() {
                 new String(channel.getQueue().get(0).array()));
     }
 
+    @Test
+    void reconnectOnMissingLogonAck() throws InterruptedException {
+        FixHandlerSettings settings = createHandlerSettings();
+        Channel channel = new Channel(settings, null);
+        FixHandler fixHandler = channel.getFixHandler();
+        fixHandler.onOpen(channel);
+        assertEquals(channel.getQueue().size(), 1);
+        assertEquals("8=FIXT.1.1\u00019=105\u000135=A\u000134=1\u000149=client\u000156=server\u000150=trader\u000152=2014-12-22T10:15:30Z\u000198=0\u0001108=30\u00011137=9\u0001553=username\u0001554=pass\u000110=203\u0001",
+            new String(channel.getQueue().get(0).array()));
+        channel.getQueue().clear();
+        Thread.sleep(settings.getLogonAckTimeout() + 2000);
```

Comment:
Why do we need such a big delay here? Maybe we could reduce it in the configuration?